spec/features/brexit_checker/question_results_spec.rb

```diff
@@ -65,6 +71,114 @@ def when_i_visit_the_brexit_checker_flow
+    and_all_business_actions_links_should_have_ecommerce_tracking
+  end
+
+  def and_business_actions_section_has_ecommerce_tracking
+    section = find('.brexit-checker-business-actions', visible: false)
+    expect(section['data-analytics-ecommerce']).to eq("")
+    expect(section['data-ecommerce-start-index']).to eq("1")
+    expect(section['data-list-title']).to eq("Brexit checker results: Your business or organisation")
+  end
+
+  def all_business_action_links_have_ecommerce_tracking(link, action_index, link_type)
+    expect(link['data-module']).to eq("track-click")
+    expect(link['data-track-category']).to eq("brexit-checker-results")
+    expect(link['data-ecommerce-path']).to eq(link['href'].sub("https://www.gov.uk",""))
+    expect(link['data-ecommerce-row']).to eq("")
+    expect(link['data-track-label']).to eq(link['href'])
+    action_string = link['data-track-action'].split(" - ")
+    expect(action_string[0]).to eql(I18n.t!("brexit_checker.results.audiences.business.heading"))
+    expect(action_string[1]).to eql(I18n.t!('brexit_checker.results.audiences.business.heading'))
+    expect(action_string[2]).to eql("1.#{action_index + 1}")
+    expect(action_string[3]).to eql(link_type)
+  end
+
+  def all_citizen_action_links_have_ecommerce_tracking(link, group_index, group_title, action_index, link_type)
+    expect(link['data-module']).to eq("track-click")
+    expect(link['data-track-category']).to eq("brexit-checker-results")
+    expect(link['data-ecommerce-path']).to eq(link['href'].sub("https://www.gov.uk",""))
+    expect(link['data-ecommerce-row']).to eq("")
+    expect(link['data-track-label']).to eq(link['href'])
+    action_string = link['data-track-action'].split(" - ")
+    expect(action_string[0]).to eql(I18n.t!("brexit_checker.results.audiences.citizen.heading"))
+    expect(action_string[1]).to eql(group_title)
+    expect(action_string[2]).to eql("#{group_index}.#{action_index + 1}")
+    expect(action_string[3]).to eql(link_type)
+  end
+
+  def and_all_business_actions_links_should_have_ecommerce_tracking
+    find_all('brexit-checker-business-actions .brexit-checker__action .brexit-checker__action_title a.govuk-link').each_with_index do |link, i|
+      all_business_action_links_have_ecommerce_tracking(link, i,"Action")
+    end
+    find_all('brexit-checker-business-actions .brexit-checker__action .govuk-body a.govuk-link').each_with_index do |link, i|
+      all_business_action_links_have_ecommerce_tracking(link, i,"Guidance")
+    end
+  end
+
+  def and_business_actions_section_has_data_search_query_attributes
+    section = find('.brexit-checker-business-actions', visible: false)
+    expect(section['data-search-query']).to eq("")
+  end
+
+  def and_all_citizens_actions_links_should_have_ecommerce_tracking
+    find_all(".brexit-checker-audience-citizen section.brexit-checker-actions__group", visible: false).each_with_index do |group, group_index|
```

Comment:
I wonder if it would be enough to check one section as all groups use the same code. This would hopefully help us simplify this test.

Reply:
As in just take the first group and run through that instead of iterating over all of them?
I wasn't sure if it was still a good idea to go over them all to account for the different variations (some with guidance only, other with action titles) some with higher group and action indexes etc?